services/skipchain/skipchain_test.go

```diff
@@ -0,0 +1,259 @@
+package skipchain
+
+import (
+	"testing"
+
+	"bytes"
+
+	"strconv"
+
+	"time"
+
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/sda"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(m *testing.M) {
+	dbg.MainTest(m)
+}
+
+func TestSkipBlock_Hash1(t *testing.T) {
+	sbd1 := NewSkipBlock()
+	sbd1.Data = []byte("1")
+	sbd1.Height = 4
+	h1 := sbd1.updateHash()
+	assert.Equal(t, h1, sbd1.Hash)
+
+	sbd2 := NewSkipBlock()
+	sbd2.Data = []byte("2")
+	sbd1.Height = 2
+	h2 := sbd2.updateHash()
+	assert.NotEqual(t, h1, h2)
+}
+
+func TestSkipBlock_Hash2(t *testing.T) {
+	local := sda.NewLocalTest()
+	hosts, el, _ := local.GenTree(2, false, false, false)
+	defer local.CloseAll()
+	sbd1 := NewSkipBlock()
+	sbd1.EntityList = el
+	sbd1.Height = 1
+	h1 := sbd1.updateHash()
+	assert.Equal(t, h1, sbd1.Hash)
+
+	sbd2 := NewSkipBlock()
+	sbd2.EntityList = local.GenEntityListFromHost(hosts[0])
+	sbd2.Height = 1
+	h2 := sbd2.updateHash()
+	assert.NotEqual(t, h1, h2)
+}
+
+func TestSkipBlockInterface(t *testing.T) {
+	// Tests the different accessors
+}
+
+func TestService_ProposeSkipBlock(t *testing.T) {
+	// First create a roster to attach the data to it
+	local := sda.NewLocalTest()
+	defer local.CloseAll()
+	_, el, service := makeHELS(local, 5)
+
+	// Setting up root roster
+	sbRoot := makeGenesisRoster(service, el)
+
+	// send a ProposeBlock
+	genesis := NewSkipBlock()
+	genesis.Data = []byte("In the beginning God created the heaven and the earth.")
+	genesis.MaximumHeight = 2
+	genesis.ParentBlockID = sbRoot.Hash
+	genesis.EntityList = sbRoot.EntityList
+	blockCount := uint32(0)
+	psbrMsg, err := service.ProposeSkipBlock(nil, &ProposeSkipBlock{nil, genesis})
+	assert.Nil(t, err)
+	psbr := psbrMsg.(*ProposedSkipBlockReply)
+	latest := psbr.Latest
+	// verify creation of GenesisBlock:
+	blockCount++
+	assert.Equal(t, blockCount, latest.Index)
+	// the genesis block has a random back-link:
+	assert.Equal(t, 1, len(latest.BackLinkIds))
+	assert.NotEqual(t, 0, latest.BackLinkIds)
+
+	next := NewSkipBlock()
+	next.Data = []byte("And the earth was without form, and void; " +
+		"and darkness was upon the face of the deep. " +
+		"And the Spirit of God moved upon the face of the waters.")
```

Comment:
And maybe more importantly that they had not suffered any kind of modifications over the years !!